### ISSUE TITLE
should loss be averaged using allreduce

### ISSUE THREAD

**Issue by LiweiPeng** [uber/horovod#482]
I am converting a single machine NMT model to use Horovod for distributed system. In the following pseudo code, should loss be averaged across all the processes using allreduce?
```
def model_fn():
    optimizer = hvd.DistributedOptimizer(optimizer)
    loss = get_loss(...)
    grad = optimizer.compte_gradients(loss)
    #Question: should loss be averaged using allreduce?
    return tf.estimator.EstimatorSpec(..., loss=loss)
```

**Comment by LiweiPeng** [comment]
I got the answer to my own question: the answer is loss should be averaged across the machines. With averaged loss, the training has better convergence.

**Comment by maxhgerlach** [comment]
@LiweiPeng I stumbled over this question by accident. Why do you believe averaging the loss helps convergence? The actual value of the loss should not play a role for the gradient descent update of the parameters IMHO.

**Comment by austinmw** [comment]
Curious if there is any benefit to doing this?
